core/services/vrf/v2/listener_v2.go

```diff
@@ -917,7 +917,7 @@ func (lsn *listenerV2) enqueueForceFulfillment(
 		requestID := common.BytesToHash(p.req.req.RequestID().Bytes())
 		subID := p.req.req.SubID()
 		requestTxHash := p.req.req.Raw().TxHash
-		etx, err = lsn.txm.CreateTransaction(txmgr.TxRequest{
+		etx, err = lsn.txm.CreateTransaction(context.Background(), txmgr.TxRequest{
```

Comment:
For these cases that don't wire in to an upstream context, should we use context.TODO()?

Reply:
That makes more sense. I went ahead and made the changes

Reply:
> For these cases that don't wire in to an upstream context, should we use context.TODO()?

Not sure what you mean by this, can you explain a little bit more?

Reply:
An orphaned Background context is problematic. Everything should be wired upstream to ensure graceful shutdown. I was thinking `TODO` as a placeholder to limit scope creep would be better than `Background` which could get lost.

But now I see that `listenerV2` already has a `utils.StopChan`, so you could just avoid the problem all together:
```go
ctx, cancel := lsn.chStop.NewCtx()
defer cancel()
etx, err = lsn.txm.CreateTransaction(ctx, txmgr.TxRequest{
```

Reply:
Totally agree with you, I posted a comment below stating that the use of TODO() isn't ideal because we should be part of the global context tree. We already have a context passed down to us from all the `processRequestsPerSub*` functions that can be piped to all called functions, but perhaps that's a big diff. Maybe the snippet you provided above works just fine.

Reply:
Thanks for the insight! I've made the updates above to the VRF changes